### PR TITLE
chore(DTFS2-7317): align docker-compose, remove docker-compose.replica-set

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates `docker-compose.gha.yml` to have the latest approaches defined in `docker-compose.replica-set.yml`

## Resolution :heavy_check_mark:
- Update `docker-compose-gha.yml`
- Remove `docker-compose.replica-set.yml`.
- Update README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,11 +93,10 @@ npm run stop
 
 ## Different docker compose files
 
-There are 3 different docker compose files right now.
+There are 2 different docker compose files right now.
 
 - docker-compose.yml is used for local development
 - docker-compose.gha.yml is used for the pipelines to run our tests against
-- docker-compose.replica-set.yml is used when you need a replica set enabled on your local database, e.g. when working with the mongodb change stream
```
Comment (ChristianMcc) [inline README.md:100]: Could we highlight that the docker-compose.gha.yml file now has the replica set enabled like this used to? (as well as the pipelines to run our tests)

File: docker-compose.gha.yml [hunk 2]
```diff
@@ -112,9 +113,10 @@ services:
       - EXTERNAL_API_KEY
       - TFM_API_URL=http://trade-finance-manager-api:5004
       - EXTERNAL_API_URL=http://external-api:5002
-      - UKEF_TFM_API_SYSTEM_KEY
+      - UKEF_TFM_API_SYSTEM_KEY=${UKEF_TFM_API_SYSTEM_KEY}
```
Comment (ChristianMcc) [inline docker-compose.gha.yml:116]: is this needed?

**Review by ChristianMcc — COMMENTED** [review]
Couple of minor comments one about the changestream.

Are we going to look at updating the mongodb versions in a separate PR?

File: docker-compose.gha.yml
Comment (MarRobSoftwire) [inline docker-compose.gha.yml:1]: Have you built using this file locally and verified that the replica set/ change stream works - worried that no one is doing this regularly with the audit logs work in testing. Happy to chat more about what this means :)

**Review by abhi-markan — DISMISSED** [review]

**Comment by abhi-markan** [comment]
Overall changes looks fine, would be keen to test this docker compose file locally to make sure MongoDB with replica set spins up on the first instance without any error. Ideally if you can start and stop this docker compose thrice to ensure no sporadic issues resides MongoDB replica set.

File: README.md (hunk 1 above)
Comment (ttbarnes) [reply README.md:100]: Good shout @ChristianMcc , updated

**Comment by ttbarnes** [comment]
Thanks @MarRobSoftwire @abhi-markan  - trying to build locally, but having docker issues.

File: docker-compose.gha.yml [hunk 3]
```diff
@@ -154,13 +157,12 @@ services:
       - AZURE_PORTAL_STORAGE_ACCOUNT
       - AZURE_PORTAL_STORAGE_ACCESS_KEY
       - AZURE_UTILISATION_REPORTS_FILESHARE_NAME
-      - UKEF_TFM_API_SYSTEM_KEY
+      - UKEF_TFM_API_SYSTEM_KEY=${UKEF_TFM_API_SYSTEM_KEY}
```
Comment (abhi-markan) [inline docker-compose.gha.yml:160]: I think this can be `UKEF_TFM_API_SYSTEM_KEY`.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

**Review by abhi-markan — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**